icu4c/source/test/cintltst/udatpg_test.c

```diff
@@ -836,4 +840,46 @@ static void TestRegionOverride(void) {
         udatpg_close(dtpg);
     }
 }
+
+static void TestISO8601(void) {
+    typedef struct TestCase {
+        const char* locale;
+        const UChar* skeleton;
+        const UChar* expectedPattern;
+    } TestCase;
+
+    const TestCase testCases[] = {
+        { "en_GB@calendar=iso8601;rg=uszzzz", u"EEEEyMMMMdjmm", u"y MMMM d, EEEE 'at' h:mm a" },
+        { "en_GB@calendar=iso8601;rg=uszzzz", u"EEEEyMMMMdHmm", u"y MMMM d, EEEE 'at' HH:mm" },
+        { "en_GB@calendar=iso8601;rg=uszzzz", u"Edjmm",         u"d, EEE, h:mm a" },
+        { "en_GB@calendar=iso8601;rg=uszzzz", u"EdHmm",         u"d, EEE, HH:mm" },
+
+        { "en_US@calendar=iso8601",           u"EEEEyMMMMdjmm", u"y MMMM d, EEEE 'at' h:mm a" },
+        { "en_US@calendar=iso8601",           u"EEEEyMMMMdHmm", u"y MMMM d, EEEE 'at' HH:mm" },
+        { "en_US@calendar=iso8601",           u"Edjmm",         u"d, EEE, h:mm a" },
+        { "en_US@calendar=iso8601",           u"EdHmm",         u"d, EEE, HH:mm" },
+
+        { "en_US",                            u"EEEEyMMMMdjmm", u"EEEE, MMMM d, y 'at' h:mm a" },
+        { "en_US",                            u"EEEEyMMMMdHmm", u"EEEE, MMMM d, y 'at' HH:mm" },
+        { "en_US",                            u"Edjmm",         u"d EEE, h:mm a" },
+        { "en_US",                            u"EdHmm",         u"d EEE, HH:mm" },
+    };
+
+    for (int32_t i = 0; i < UPRV_LENGTHOF(testCases); i++) {
+        UErrorCode err = U_ZERO_ERROR;
+        UDateTimePatternGenerator* dtpg = udatpg_open(testCases[i].locale, &err);
+
+        if (assertSuccess("Error creating dtpg", &err)) {
+            UChar actualPattern[200];
+
+            udatpg_getBestPatternWithOptions(dtpg, testCases[i].skeleton, -1, 0, actualPattern, 200, &err);
```

Comment:
See my review comments on the previous PR #3103: Please don't hardcode lengths, that becomes increasingly brittle when code changes over time, use UPRV_LENGTHOF(actualPattern) instead of 200.

Reply:
I agree (though if Peter has limited time to fix this PR, it could be fixed later.)

Reply:
I think you're both being overly picky, but I made the change.

Reply:
You already made those changes in PR #3103 so even if you don't think it's particularly important I just don't understand why you'd prefer to throw those improvements away instead of simply copy-pasting that improved code into this PR.

Reply:
I think I changed it over there because somebody (probably you) complained there.  I didn't take it out intentionally; I started over from scratch because I didn't remember I had that other PR still open.  Regardless, the fix is back in now...

Reply:
I see you went ahead and made this change on your own.  I thought I had made it in my last commit; but maybe I forgot that file.